### ISSUE TITLE
dns resolution doesn't seem to work for ec2.internal addresses

### ISSUE THREAD

**Issue by dustymabe** [openshift/origin#11962]
I believe this is related to https://github.com/golang/go/issues/17967

The go resolver in openshift seems to not be able to resolve ec2 internal hostnames and is causing issues when trying to start the openshift master containerized on a Fedora 25 atomic host. I basically end up getting a bunch of errors like this:

```
E1117 19:36:30.711042       1 reflector.go:214] github.com/openshift/origin/vendor/k8s.io/kubernetes/plugin/pkg/admission/resourcequota/resource_access.go:83: Failed to list *api.ResourceQuota: Get https://ip-10-0-164-174.ec2.internal:8443/api/v1/resourcequotas?resourceVersion=0: dial tcp: lookup ip-10-0-164-174.ec2.internal on 10.0.0.2:53: no such host
E1117 19:36:30.711069       1 reflector.go:203] github.com/openshift/origin/vendor/k8s.io/kubernetes/plugin/pkg/admission/limitranger/admission.go:154: Failed to list *api.LimitRange: Get https://ip-10-0-164-174.ec2.internal:8443/api/v1/limitranges?resourceVersion=0: dial tcp: lookup ip-10-0-164-174.ec2.internal on 10.0.0.2:53: no such host
```

I am able to workaround this issue if I manually set it so that it queries 10.0.164.174.xip.io instead. 

##### Version
```
# openshift version
openshift v1.4.0-alpha.1+f189ede
kubernetes v1.4.0+776c994
etcd 3.1.0-rc.0
```

##### Steps To Reproduce
1. Run the openshift byo installer against an aws machine (I chose F25 atomic host, ami-c4052cd3) . Here is the inventory file I used:
```
# Create an OSEv3 group that contains the masters and nodes groups
[OSEv3:children]
masters
nodes
etcd

# Set variables common for all OSEv3 hosts
[OSEv3:vars]
ansible_user=fedora
ansible_become=true
deployment_type=origin
containerized=true
openshift_release=v1.3.1

# host group for masters
[masters]
ec2-54-147-83-147.compute-1.amazonaws.com openshift_public_hostname=ec2-54-147-83-147.compute-1.amazonaws.com

# host group for nodes
[nodes]
ec2-54-147-83-147.compute-1.amazonaws.com openshift_public_hostname=ec2-54-147-83-147.compute-1.amazonaws.com

# host group for etcd
[etcd]
ec2-54-147-83-147.compute-1.amazonaws.com openshift_public_hostname=ec2-54-147-83-147.compute-1.amazonaws.com
```

2. Here is the command I ran: `ansible-playbook -i myinventory playbooks/byo/config.yml -e 'ansible_python_interpreter=/usr/bin/python3' -v`

##### Current Result
openshift master does not come up

##### Expected Result
openshift master is able to resolve ec2.internal address and come up

**Comment by dustymabe** [comment]
@sdodson and @dcbw helped me debug this issue and @dcbw ultimately reported the issue upstream in go at https://github.com/golang/go/issues/17967. Thank you both for helping me debug today.

**Comment by dustymabe** [comment]
hit this again today - not sure if we are able to work around this or if we have to wait for go 1.8 first.

**Comment by dustymabe** [comment]
haven't tested this but it should be fixed now since origin is compiled on go 1.8+
